src/careamics/models/lvae/likelihoods.py

```diff
@@ -104,83 +114,64 @@ def forward(
         p(x|z_1) = N(x|\mu_{p,1}, \sigma_{p,1}^2)
     """
 
     def __init__(
         self,
-        ch_in: int,
-        color_channels: int,
-        predict_logvar: Literal[None, "pixelwise"] = None,
+        predict_logvar: Union[Literal["pixelwise"], None] = None,
         logvar_lowerbound: Union[float, None] = None,
-        conv2d_bias: bool = True,
     ):
-        """
-        Constructor.
+        """Constructor.
 
         Parameters
         ----------
-        predict_logvar: Literal[None, 'global', 'pixelwise', 'channelwise'], optional
-            If not `None`, it expresses how to compute the log-variance.
-            Namely:
-            - if `pixelwise`, log-variance is computed for each pixel.
-            - if `global`, log-variance is computed as the mean of all pixel-wise entries.
-            - if `channelwise`, log-variance is computed as the average over the channels.
-            Default is `None`.
+        predict_logvar: Union[Literal["pixelwise"], None], optional
+            If `pixelwise`, log-variance is computed for each pixel, else log-variance
+            is not computed. Default is `None`.
         logvar_lowerbound: float, optional
             The lowerbound value for log-variance. Default is `None`.
-        conv2d_bias: bool, optional
-            Whether to use bias term in convolutions. Default is `True`.
         """
         super().__init__()
 
-        # If True, then we also predict pixelwise logvar.
         self.predict_logvar = predict_logvar
         self.logvar_lowerbound = logvar_lowerbound
-        self.conv2d_bias = conv2d_bias
         assert self.predict_logvar in [None, "pixelwise"]
 
-        # logvar_ch_needed = self.predict_logvar is not None
-        # self.parameter_net = nn.Conv2d(ch_in,
-        #                                color_channels * (1 + logvar_ch_needed),
-        #                                kernel_size=3,
-        #                                padding=1,
-        #                                bias=self.conv2d_bias)
-        self.parameter_net = nn.Identity()
-
         print(
             f"[{self.__class__.__name__}] PredLVar:{self.predict_logvar} LowBLVar:{self.logvar_lowerbound}"
         )
 
-    def get_mean_lv(self, x: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+    def get_mean_lv(self, x: torch.Tensor) -> tuple[torch.Tensor, torch.Tensor]:
         """
         Given the output of the top-down pass, compute the mean and log-variance of the
         Gaussian distribution defining the likelihood.
 
         Parameters
         ----------
         x: torch.Tensor
-            The input tensor to the likelihood module, i.e., the output of the top-down pass.
+            The input tensor to the likelihood module, i.e., the output
+            the LVAE 'output_layer'. Shape is: (B, 2 * C, [Z], Y, X) in case
+            `predict_logvar` is not None, or (B, C, [Z], Y, X) otherwise.
```

Comment:
So just to clarify, you have to know the value of the attribute `predict_logvar` to know what dimensions of `x` are valid for this method...?

(Not to be addressed now but we should make a note to refactor this, especially since, now that `parameter_net` is always the identity the mean is not altered at all by this method.)

Reply:
So yeah, it seems if the options for `"channelwise"` and `"global"` are removed, then this function basically does nothing anymore, apart from optionally clip the logvar.

Reply:
Yes, `predict_logvar` determines the number of channels in `x`. Moreover, `parameter_net` is not used anymore here, since it was moved into the LVAE `nn.Module` (which makes waaaaay more sense)